UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -1,587 +1,1316 @@
+			var vexinvY = (1d / (direction.y)); //Needs to be conditional
+
+			double calculationFloat = 0;
+
+			bool LeftFaceHit = true;
+
+			while (Math.Abs(RelativeX) < Mathf.Abs(Relativetarget.x) ||
+			       Math.Abs(RelativeY) < Mathf.Abs(Relativetarget.y))
+				//for (int Ai = 0; Ai < 6; Ai++)
+			{
+				//if (xBuildUp > yBuildUp)
+				if ((xSteps + gridOffsetx + stepX) * vexinvX < (ySteps + gridOffsety + stepY) * vexinvY
+				) // which one has a lesser multiplication factor since that will give a less Magnitude
+				{
+					xSteps += stepX;
+
+					calculationFloat = ((xSteps + gridOffsetx) * vexinvX);
+
+					RelativeX = direction.x * calculationFloat; //Remove offset here maybe?
+					RelativeY = direction.y * calculationFloat;
+
+					LeftFaceHit = true;
+				}
+				//else if (xBuildUp < yBuildUp)
+				else //if (xBuildUp < yBuildUp)
+				{
+					ySteps += stepY;
+					calculationFloat = ((ySteps + gridOffsety) * vexinvY);
+
+					RelativeX = direction.x * calculationFloat;
+					RelativeY = direction.y * calculationFloat;
+
+					LeftFaceHit = false;
+				}
+
+
+				vec.x = (int) Mathf.Round(origin.x) + xSteps;
+				vec.y = (int) Mathf.Round(origin.y) + ySteps;
+
+				vecHit.x = origin.x + (float) RelativeX; //+ offsetX;
+				vecHit.y = origin.y + (float) RelativeY; // + offsetY;
+				//Check point here
+
+				if (LeftFaceHit)
+				{
+					// float TestX = ((vecHit.x - 0.5f) - Mathf.Floor(vecHit.x));
+
+					// if (0.05f < Math.Abs(TestX))
+					// {
+						// Logger.Log("Offsetuntouchx = " + Offsetuntouchx + "\n" + "directionx = " + direction.x + "\n" +
+						           // "Step = " + xSteps + "\n" + "vexinv = " + vexinvX + "\n" + "offset = " + offsetX +
+						           // "\n" + "\n" + "Test =" + TestX + "\n" + "Relative =" + (RelativeX)
+						           // + "\n" + "\n"
+						           // + " direction.x = " +  direction.x + " calculationFloat " + calculationFloat
+						           // + "\n" + " xSteps " + xSteps + " gridOffsetx " + gridOffsetx  +"  vexinvX " + vexinvX);
+
+
+					// }
+				}
+				else
+				{
+					// float Testy = ((vecHit.y - 0.5f) - Mathf.Floor(vecHit.y));
+					// if (0.05f < Math.Abs(Testy))
+					// {
+						// Logger.Log("Offsetuntouchx = " + Offsetuntouchy + "\n" + "directionx = " + direction.y + "\n" +
+						           // "Step = " + ySteps + "\n" + "vexinv = " + vexinvY + "\n" + "offset = " + offsetY +
+						           // "\n" + "\n" + "Test =" + Testy + "\n" + "Relative =" + (RelativeY)
+						           // + "\n" + "\n"
+						           // + " direction.y = " +  direction.y + " calculationFloat " + calculationFloat
+						           // + "\n" + " ySteps " + ySteps + " gridOffsety " + gridOffsety  +"  vexinvY " + vexinvY);
+					// }
+				}
+
+				for (var i = 0; i < LayersValues.Length; i++)
+				{
+					if (LayersValues[i].LayerType == LayerType.Objects) continue;
+					if (LTSUtil.IsLayerIn(layerMask, LayersValues[i].LayerType))
+					{
+						TileLcation = null;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'TileLcation'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6319358)